pkg/p2p/wire/message/message.go

```diff
@@ -103,7 +110,12 @@ func (m simple) Equal(other Message) bool {
 
 // New creates a new Message
 func New(t topics.Topic, payload interface{}) Message {
-	return &simple{category: t, payload: payload}
+	return &simple{category: t, payload: payload, header: nil}
+}
+
+// NewWithHeader creates a new Message with non-nil header
+func NewWithHeader(t topics.Topic, payload interface{}, header []byte) Message {
```

Comment:
I would prefer a single New constructor, if there is a way in golang to assign default argument value.  To avoid the need to append `nil` to all 60 `message.New` that does not need header, I chose a second constructor.

Reply:
Fair